etc/run_in_headless_android_emulator.py

```diff
@@ -55,10 +55,10 @@ def main(avd_name, apk_path, *args):
         args = list(args)
         write_user_stylesheets(adb, args)
         write_hosts_file(adb)
-        write_args(adb, args)
 
-        check_call(adb + ["shell", "am start com.mozilla.servo/com.mozilla.servo.MainActivity"],
-                   stdout=sys.stderr)
+        cmd_args = "-e servoargs \'" + " ".join(args) + "\'" if args else ""
```

Comment:
Joining with spaces like this breaks the case where one of the argument contains a space. Could we use one of these instead? (From `adb shell am -h`)

```
    [--esa <EXTRA_KEY> <EXTRA_STRING_VALUE>[,<EXTRA_STRING_VALUE...]]
        (mutiple extras passed as String[]; to embed a comma into a string,
         escape it using "\,")
    [--esal <EXTRA_KEY> <EXTRA_STRING_VALUE>[,<EXTRA_STRING_VALUE...]]
        (mutiple extras passed as List<String>; to embed a comma into a string,
         escape it using "\,")
```

And `shell_quote` should still be used.